services/controllers/tickers/base.go

```diff
@@ -34,34 +36,159 @@ func NewController(
 	}
 }
 
-func (c Controller) HandleTickersRequestV2(tr controllers.TickerRequestV2) (controllers.TickerResponseV2, error) {
-	rate, err := c.getRateByPriority(strings.ToUpper(tr.Currency))
+func (c Controller) HandleTickersRequest(request controllers.TickerRequest) (watchmarket.Tickers, error) {
+	rate, err := c.getRateByPriority(strings.ToUpper(request.Currency))
 	if err != nil {
-		return controllers.TickerResponseV2{}, errors.New(watchmarket.ErrNotFound)
+		return watchmarket.Tickers{}, errors.New(watchmarket.ErrNotFound)
 	}
 
-	tickers, err := c.getTickersByPriority(makeTickerQueriesV2(tr.Ids))
+	tickers, err := c.getTickersByPriority(request.Assets)
 	if err != nil {
-		return controllers.TickerResponseV2{}, errors.New(watchmarket.ErrInternal)
+		return watchmarket.Tickers{}, errors.New(watchmarket.ErrInternal)
 	}
+	tickers = c.filterTickers(tickers, rate)
+	return tickers, nil
+}
 
-	tickers = c.normalizeTickers(tickers, rate)
-
-	return createResponseV2(tr, tickers), nil
+func (c Controller) filterTickers(tickers watchmarket.Tickers, rate watchmarket.Rate) (result watchmarket.Tickers) {
+	for _, ticker := range tickers {
+		r, ok := c.rateToDefaultCurrency(ticker, rate)
+		if !ok {
+			continue
+		}
+		if !watchmarket.IsSuitableUpdateTime(ticker.LastUpdate, c.configuration.RestAPI.Tickers.RespectableUpdateTime) {
+			continue
+		}
+		c.applyRateToTicker(&ticker, r)
+		result = append(result, ticker)
+	}
+	return result
 }
 
-func (c Controller) HandleTickersRequest(tr controllers.TickerRequest) (controllers.TickerResponse, error) {
-	rate, err := c.getRateByPriority(strings.ToUpper(tr.Currency))
+func (c Controller) getTickersByPriority(assets []controllers.Asset) (watchmarket.Tickers, error) {
+	if c.configuration.RestAPI.UseMemoryCache {
```

Comment:
We always have `UseMemoryCache` as true, I think we can remove anything in `else`

Reply:
we can even remove `UseMemoryCache` property as it's always in memory